pkg/internal/regressiontests/internal/typescript/src/authfetch/auth-fetch-adapter.ts

```
@@ -0,0 +1,91 @@
+import {AuthFetch} from "@bsv/sdk";
+import {Config, FetchResponse} from "../gen/auth_fetch";
+import {Show} from "../show/show";
+
+export interface AuthFetchRequest {
+    url: string;
+    config?: Config;
+}
+
+export class AuthFetchAdapter {
+    constructor(private show: Show, private authFetch: AuthFetch) {
+    }
+
+    async fetch(req: AuthFetchRequest): Promise<FetchResponse> {
+        let url = this.extractUrl(req);
+
+        const {config} = req;
+        let {retryCounter} = config || {}
+        if (!retryCounter) {
+            retryCounter = 1
+        }
+
+
+        try {
+            const response = await this.authFetch.fetch(
+                url,
+                {
+                    method: config?.method || undefined,
```

Is the `|| undefined` necessary? I guess, you want to make it `undefined` when empty string or `null` is provided

---

yes, it's necessary, otherwise it is set as null (or empty string - I don't remember) because this is how the grpc is working, while AuthFetch has specifically check for undefined. Without this line the calls will fail, because it will use empty method, instead of default `GET`.